File: SDK/Service/ModelService/APIs/StaticDataApi.cs
using System.Collections.Generic;
using Microsoft.Extensions.Logging;
using Raid.Service.DataModel;

namespace Raid.Service
{
    [PublicApi("static-data")]
    internal class StaticDataApi : ApiHandler
    {
        private StaticDataCache StaticDataCache;
        public StaticDataApi(ILogger<ApiHandler> logger, StaticDataCache staticData)
            : base(logger) => StaticDataCache = staticData;

        [PublicApi("getAllData")]
        public StaticData GetAllData()
        {
            return StaticDataFacet.ReadValue(StaticDataCache);
        }

        [PublicApi("getLocalizedStrings")]
        public IReadOnlyDictionary<string, string> GetLocalizedStrings()
        {
            return StaticDataFacet.ReadValue(StaticDataCache).LocalizedStrings;
        }

        [PublicApi("getArenaData")]
        public StaticArenaData GetArenaData()
        {
            return StaticDataFacet.ReadValue(StaticDataCache).ArenaData;
        }

        [PublicApi("getArtifactData")]
        public StaticArtifactData GetArtifactData()
        {
            return StaticDataFacet.ReadValue(StaticDataCache).ArtifactData;
        }

        [PublicApi("getHeroData")]
        public StaticHeroData GetHeroData()
        {
            return StaticDataFacet.ReadValue(StaticDataCache).HeroData;
        }

        [PublicApi("getSkillData")]
        public StaticSkillData GetSkillData()
        {
            return StaticDataFacet.ReadValue(StaticDataCache).SkillData;
        }

        [PublicApi("getStageData")]
        public StaticStageData GetStageData()
        {
            return StaticDataFacet.ReadValue(StaticDataCache).StageData;
        }
    }
}

File: SDK/Python/src/raidtoolkit/StaticDataApi.py
from .RemoteApiClient import RemoteApiClient


class StaticDataApi:
    def __init__(self, client: RemoteApiClient):
        self.client = client

    def get_all_data(self):
        return self.client.call("static-data", "getAllData")

    def get_localized_strings(self):
        return self.client.call("static-data", "getLocalizedStrings")

    def get_arena_data(self):
        return self.client.call("static-data", "getArenaData")

    def get_artifact_data(self):
        return self.client.call("static-data", "getArtifactData")

    def get_hero_data(self):
        return self.client.call("static-data", "getHeroData")

    def get_skill_data(self):
        return self.client.call("static-data", "getSkillData")

    def get_stage_data(self):
        return self.client.call("static-data", "getStageData")
